File: server/SprintBoard.Api/Models/Issue.cs
namespace SprintBoard.Api.Models;

public class Issue
{
    public int Id { get; set; }

    // Project relationship
    public int ProjectId { get; set; }
    public Project Project { get; set; } = null!;

    // Basic issue information
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }

    // State
    public IssueStatus Status {get; set;}

    // Priority
    public Priority Priority { get; set; } = Priority.Medium;

    // Assignment
    public int? AssigneeId { get; set; }
    public User? Assignee { get; set; }

    // Creation
    public int CreatedById { get; set; }
    public User CreatedBy { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Optional lifecycle information
    public DateTime? UpdatedAt { get; set; }
}

File: client/src/types/Issue.ts
// Adjust these if your backend serializes enums as numbers instead of strings
// (i.e. if you're not using JsonStringEnumConverter in SprintBoard.Api).
export type Priority = "Low" | "Medium" | "High";
export type IssueStatus = "Todo" | "InProgress" | "InReview" | "Done";

export interface Assignee {
  id: number;
  name: string;
  avatarUrl?: string | null;
}

// Mirrors IssueResponseDto
export interface Issue {
  id: number;
  name: string;
  description?: string | null;
  priority: Priority;
  status: IssueStatus;
  assignee?: Assignee | null;
  createdAt: string;
  updatedAt?: string | null;
}

// Mirrors CreateIssueDto
export interface CreateIssuePayload {
  name: string;
  description?: string | null;
  priority: Priority;
  assigneeId?: number | null;
}

// Mirrors UpdateIssueDto
export interface UpdateIssuePayload {
  name: string;
  description?: string | null;
  priority: Priority;
  status: IssueStatus;
  assigneeId?: number | null;
}
